ios/RNVoipPushNotification/RNVoipPushNotificationManager.m

```diff
@@ -30,27 +30,28 @@ @implementation RNVoipPushNotificationManager
 static NSString *_lastVoipToken = @"";
 static NSMutableDictionary<NSString *, RNVoipPushNotificationCompletion> *completionHandlers = nil;
 
-
 // =====
 // ===== RN Module Configure and Override =====
 // =====
 
+-(instancetype) init {
+    return [[self class] sharedInstance];
+}
 
-- (instancetype)init
-{
-    if (self = [super init]) {
-        _delayedEvents = [NSMutableArray array];
-    }
-    return self;
+-(instancetype) initPrivate {
+    _delayedEvents = [NSMutableArray array];
+    return [super init];
```

Comment:
wouldn't it be:

```objc
- (instancetype)initPrivate
{
    if (self = [super init]) {
        _delayedEvents = [NSMutableArray array];
    }
    return self;
}
```

?

Reply:
The diff might looks like this:

```diff
- (instancetype)init
+ (instancetype)initPrivate
```

all looks good now except this.

Reply:
There are two methods init and initPrivate. Init should be there as it is inherited from base class and it invocation will return allocated and initialised only once with initPrivate method.
It is based on https://stackoverflow.com/a/7035136/3686678

Reply:
> wouldn't it be:
> 
> ```objc
> - (instancetype)initPrivate
> {
>     if (self = [super init]) {
>         _delayedEvents = [NSMutableArray array];
>     }
>     return self;
> }
> ```
> 
> ?

Actually initPrivate is called only once so it should be ok.

Reply:
yeh, I know, I just want to keep it as less diff as possible, and not cause too much confuses for the future developers

Reply:
Yep, I see. But I also prefer to improve code ). And for me is not appropriate to make an assignment inside condition. It is unusual and hard to read. I would better go with extra line for  assignment and the check in if condition.